crates/sui-core/src/authority.rs

```diff
@@ -772,6 +811,24 @@ impl AuthorityState {
             .tap_err(|e| debug!(?tx_digest, "process_certificate failed: {e}"))
     }
 
+    pub async fn notify_read_transaction_info(
+        &self,
+        certificate: &VerifiedCertificate,
+    ) -> SuiResult<VerifiedTransactionInfoResponse> {
+        let tx_digest = *certificate.digest();
+        let effects = self
+            .database
+            .notify_read_effects(vec![tx_digest])
+            .await?
+            .pop()
+            .unwrap();
```

Comment:
is this unwrap safe?

Reply:
Added a comment in the `expect()` call. `notify_read_effects()` with 1 element input vector should return exact 1 element output vector, so the unwrap() on pop() is safe.

Reply:
It turns out before checkpoint v2 can help fullnode catch up, TransactionOrchestrator needs to initiate fetching missing parents via node sync. This behavior is still tested. I updated the comment to replace this function after checkpoint v2.